### PR TITLE
feat(portals): establish registry and host context

### PR DESCRIPTION
**Pull request by Connorbelez**
feat(portals): establish registry and host context

responding to feedback

specs

## Summary by Sourcery

Introduce a persisted portal registry and root-level host resolution context, wiring it through backend migrations, schema, and the frontend router to classify hosts and scope data access per portal.

New Features:
- Add a Convex portal registry schema, including portal documents, attachment tables, and per-user home portal references.
- Introduce shared portal host/slug contracts and helpers for marketing, admin, and broker/FairLend hosts across backend and frontend.
- Add root-level portal context resolution and cache scoping in the router so requests are classified by host before child routes run.

Enhancements:
- Extend broker migrations to seed the FairLend portal, backfill broker portal rows, and assign deterministic home portals to users.
- Add a fail-closed UI boundary and updated home route that render based on resolved portal state instead of ad hoc host parsing.
- Scope React Query caching by resolved portal context to avoid cross-portal data leakage.

Tests:
- Add Convex tests for portal registry backfill, host-based lookup, and duplicate-claim invariants.
- Add frontend tests for host extraction, portal context resolution, portal-state boundary behavior, and root-route blocking of invalid hosts.
- Add tests for portal-scoped query cache hashing and update execution/spec audit artifacts for ENG-297.

## Summary by CodeRabbit

* **New Features**
  * Implemented persistent portal registry with host-based resolution and classification (marketing, admin, reserved, unknown).
  * Added portal state boundary that blocks access to unavailable or reserved portals with appropriate error messaging.
  * Introduced portal-scoped query caching to optimize user experience across different portal contexts.
  * Users are now automatically assigned to their appropriate home portal based on broker/lender/borrower relationships.

### CODE REVIEW

File: convex/brokers/migrations.ts [hunk 1]
```diff
@@ -149,3 +466,72 @@ export const getOrgScopeBackfillStatus = adminQuery
...
+		for (const broker of brokers) {
+			if (!broker.orgId) {
+				brokersMissingOrgIdCount += 1;
+				continue;
+			}
+
+			const brokerPortal = await getPortalByBrokerId(ctx, broker._id);
+			if (!brokerPortal) {
+				brokersMissingPortalCount += 1;
+			}
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/brokers/migrations.ts]: **Fix:** `getPortalRegistryBackfillStatus` no longer calls `getPortalByBrokerId` per broker. It builds `brokerIdsWithPortal` from the single `portals` collect, and derives FairLend staff admin `authId`s with one `organizationMemberships` query on `byOrganization` + `FAIRLEND_STAFF_ORG_ID` instead of `isFairLendStaffAdminUser` per user.

File: specs/ENG-297/chunks/chunk-01-portal-schema/context.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+## Existing code touchpoints
+- `convex/schema.ts`
+- `convex/brokers/migrations.ts`
+- `convex/lib/orgScope.ts`
+- `convex/fluent.ts`
+- existing broker, lender, borrower, and onboarding schema relationships
+
+## Validation
+- `bunx convex codegen`
+- `bunx vitest run convex/portals/__tests__/lookup.test.ts convex/portals/__tests__/migrations.test.ts`
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply specs/ENG-297/chunks/chunk-01-portal-schema/context.md]: **Fix:** Validation section now points at `bun run test -- convex/portals/__tests__/registry.test.ts` (the consolidated portal registry tests).
